climada/util/coordinates.py

```diff
@@ -1155,23 +1158,81 @@ def get_admin1_info(country_names):
                                             name='admin_1_states_provinces')
     admin1_recs = shapefile.Reader(admin1_file)
     admin1_info = dict()
     admin1_shapes = dict()
-    for iso3 in country_names:
-        admin1_info[iso3] = list()
-        admin1_shapes[iso3] = list()
+    for country in countries:
+        if isinstance(country, (int, float)):
+            country = f'{int(country):03d}' # transform numerric code to str
+        country = pycountry.countries.lookup(country).alpha_3 # get iso3a code
+        admin1_info[country] = list()
+        admin1_shapes[country] = list()
         for rec, rec_shp in zip(admin1_recs.records(), admin1_recs.shapes()):
-            if rec['adm0_a3'] == iso3:
-                admin1_info[iso3].append(rec)
-                admin1_shapes[iso3].append(rec_shp)
+            if rec['adm0_a3'] == country:
+                admin1_info[country].append(rec)
+                admin1_shapes[country].append(rec_shp)
     return admin1_info, admin1_shapes
 
+def get_admin1_geometries(countries):
+    """
+    return geometries, names and codes of admin 1 regions in given countries
+    in a GeoDataFrame. If no admin 1 regions are defined, all regions in countries
+    are returned.
+
+    Parameters
+    ----------
+    countries : list or str
+        string or list with strings, either ISO3 code or names of countries, e.g.:
+        ['ZWE', 'GBR', 'VNM', 'UZB', 'Kenya', '051']
+        For example, for Armenia, the following inputs work:
+            'Armenia', 'ARM', '051', 51
+
+    Returns
+    -------
+    gdf : GeoDataFrame
+        geopandas.GeoDataFrame instance with columns:
+            "admin1_name" : str
+                name of admin 1 region
+            "iso_3166_2" : str
+                iso code of admin 1 region
+            "geometry" : Polygon or MultiPolygon
+                shape of admin 1 region as shapely geometry object
+            "iso_3n" : str
+                numerical iso 3 code of country (admin 0)
+            "iso_3a" : str
+                alphabetical iso 3 code of country (admin 0)
+    """
+    # init empty GeoDataFrame:
+    gdf = gpd.GeoDataFrame(
+        columns = ("admin1_name", "iso_3166_2", "geometry", "iso_3n", "iso_3a"))
+
+    # extract admin 1 infos and shapes for each country:
+    admin1_info, admin1_shapes = get_admin1_info(countries)
+    for country in admin1_info.keys():
+        # fill admin 1 region names and codes to GDF for single country:
+        gdf_tmp = gpd.GeoDataFrame(columns = gdf.columns)
+        gdf_tmp.admin1_name = [record['name'] for record in admin1_info[country]]
+        gdf_tmp.iso_3166_2 = [record['iso_3166_2'] for record in admin1_info[country]]
+        geoseries = gpd.GeoSeries()
+        for shape in admin1_shapes[country]:
+            # fill shape into GeoSeries,
+            # this automatically transforms shape to Polygon object:
+            geoseries = geoseries.append(gpd.GeoSeries(shape))
```

Comment:
could this be a list comprehension?
```suggestion
       geoseries = gpd.GeoSeries([gpd.Geoseries(shape) for shape in admin1_shapes[country]])
```